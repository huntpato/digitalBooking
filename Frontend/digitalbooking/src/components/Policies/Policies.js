import React from 'react'
import styles from './Policies.module.css'

export const Policies = (props) => {
    const {rules, rulesInfo, houseRules, healthSecurity,cancellation} = styles;
   const {hrules, health, cancel} = props;
    console.log(hrules);
    return (
        <section className={rules}>
          <h4>Qué tenés que saber</h4>
          <hr/>
          <div className={rulesInfo}>
            <div className={houseRules}>
              <h4>Normas de la casa</h4>
              <ul>
                {hrules && hrules.split(",").map((rule)=><li>{rule}</li> )}
              </ul>
            </div>
            <div className={healthSecurity}>
              <h4>Salud y seguridad</h4>
              <ul>
                {health && health.split(",").map((rule)=><li>{rule}</li> )}
              </ul>
            </div>
            <div className={cancellation}>
              <h4>Política de cancelación</h4>
              <ul>
              {cancel && cancel.split(",").map((rule)=><li>{rule}</li> )}
                <li>Los pagos con tarjeta de crédito son reembolsables.</li>
              </ul>
            </div>
          </div>
        </section>
    )
}
