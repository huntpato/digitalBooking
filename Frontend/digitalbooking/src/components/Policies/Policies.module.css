/*rules section*/

.rules {
  align-self: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
  margin: 20px 0px 20px 0px;
  width: 90vw;
}

.rules h4 {
  color: var(--color-typo);
  margin-bottom: 10px;
  font-size: 18px;
}

.rules hr {
  background-color: var(--principal-color);
  border: none;
  width: 100%;
  height: 1.4px;
  margin-bottom: 40px;
}

.rules ul {
  padding: 0px;
}

.rules li {
  list-style-type: none;
  line-height: 22px;
  font-size: 14px;
  margin-bottom: 18px;
  width: 95%;
}

.rulesInfo {
  align-items: flex-start;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  width: 100%;
}

.houseRules {
  width: 25%;
}

.houseRules h4 {
  color: var(--color-typo);
  margin-bottom: 30px;
}

.healthSecurity {
  width: 25%;
}

.healthSecurity h4 {
  color: var(--color-typo);
  margin-bottom: 30px;
}

.cancellation {
  width: 25%;
}

.cancellation h4 {
  color: var(--color-typo);
  margin-bottom: 30px;
}

@media (max-width: 768px) {
  /*rules section*/

  .rulesInfo {
    flex-wrap: wrap;
  }

  .houseRules {
    width: 50%;
    margin-bottom: 40px;
  }

  .healthSecurity {
    width: 50%;
    margin-right: 0%;
    margin-bottom: 40px;
  }

  .cancellation {
    width: 50%;
  }
}

@media (max-width: 480px) {
  .rulesInfo {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  .houseRules {
    width: 100%;
    margin-bottom: 20px;
  }

  .healthSecurity {
    width: 100%;
    margin-bottom: 20px;
  }

  .cancellation {
    width: 100%;
  }
}
