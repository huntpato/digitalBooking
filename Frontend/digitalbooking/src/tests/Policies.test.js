import React from 'react';
import { shallow } from 'enzyme';
import { Policies } from '../components/Policies/Policies'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <Policies/>', ()=> {

    const wrapper= shallow(<Policies hrules="house rules" health="health" cancel="cancel"/>);

    test('Comprobar que <Policies/> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const main = wrapper.find('section');
        expect(main.hasClass('rules')).toBe(true);
    })
    
    test('Verificar que las props renderinzan correctamente', () => {

        const rules = wrapper.find('li').at(0);
        expect(rules.text()).toContain('house rules');
        
        const h = wrapper.find('li').at(1);
        expect(h.text()).toContain('health');
        
        const c = wrapper.find('li').at(2);
        expect(c.text()).toContain('cancel');
    })
    

})
