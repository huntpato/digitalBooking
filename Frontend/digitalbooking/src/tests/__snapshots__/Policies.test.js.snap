// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <Policies/> Comprobar que <Policies/> renderiza correctamente. 1`] = `
<section
  className="rules"
>
  <h4>
    Qué tenés que saber
  </h4>
  <hr />
  <div
    className="rulesInfo"
  >
    <div
      className="houseRules"
    >
      <h4>
        Normas de la casa
      </h4>
      <ul>
        <li>
          house rules
        </li>
      </ul>
    </div>
    <div
      className="healthSecurity"
    >
      <h4>
        Salud y seguridad
      </h4>
      <ul>
        <li>
          health
        </li>
      </ul>
    </div>
    <div
      className="cancellation"
    >
      <h4>
        Política de cancelación
      </h4>
      <ul>
        <li>
          cancel
        </li>
        <li>
          Los pagos con tarjeta de crédito son reembolsables.
        </li>
      </ul>
    </div>
  </div>
</section>
`;
